index.js

```diff
@@ -6,46 +6,55 @@
 'use strict';
 module.exports = leftPad;
 
-var cache = [
-  '',
-  ' ',
-  '  ',
-  '   ',
-  '    ',
-  '     ',
-  '      ',
-  '       ',
-  '        ',
-  '         '
-];
+// Generate cache. Strings are UTF16 in JavaScript,
```

Comment:
So, the length 10 was picked arbitrarily after looking at the most common cases. It helps significantly in very common cases (aligning with spaces), but I don't feel too strongly about it.

Also, note the engine has no obligation to store these strings as 528*2 bytes.

Reply:
You're completely right; I recall reading a mozilla blog post about the switch to faster, leaner data structures (some kind of rope, presumably). Speculating: in that case generating, the cached strings might also ensure better use of those internal data structures, making it a good idea regardless of what size it becomes.

Still, when in doubt I prefer to assume worst-case; like for people stuck with older browsers

Reply:
I'll clarify, the whole "optimization" of left pad that was done was for Node (where it might run a lot in build systems and servers) is _specifically_ based on the fact V8 represents strings with trees and collapses them to character arrays when it needs to - and the construction of the tree collapses anyway.

You can `padStart` and `padEnd` with modern versions anyway - I wanted to see if collapsing improved it and by how much :D

Reply:
> I'll clarify, the whole "optimization" of left pad that was done was for Node (where it might run a lot in build systems and servers) 

Aaah, ok. 

A bit of background info: I have been programming for decades, but I only started looking at JS in the last two years, and even then in complete isolation on a very specific niche SPA for biologists. Since they use python on the backend, because SciPy, I have not used Node at all in that time (`npm` doesn't really count, I think). So I have huge gaps in contextual understanding of how the code is used in practice.

If you have any more feedback on what kind of optimisation matters, let me know.